src/my-workspace/views/MyWorkspace.tsx

```diff
@@ -1,52 +1,172 @@
+				label: 'Auteur',
+				options: [
+					{ id: 'all', label: 'Alles' },
+					{ id: 'owner', label: 'Enkel waar ik eigenaar ben' },
+					{ id: 'sharedWith', label: 'Enkel gedeeld met mij' },
+					{ id: 'sharedBy', label: 'Enkel gedeeld door mij' },
+				],
+			},
+		},
+		[MAPS_ID]: {
+			amount: getAmount([]),
+			component: <span>TODO Mappen</span>,
+			filter: {
+				label: 'Filter op label',
+				options: [{ id: 'all', label: 'Alle' }],
+			},
+		},
+		[BOOKMARKS_ID]: {
+			addHandler: () => {},
+			amount: getAmount([]),
+			component: <Bookmarks />,
+		},
+		[FAVORITES_ID]: {
+			amount: getAmount([]),
+			component: <span>TODO Favorieten</span>,
+		},
+	};
+	// Set active tab based on above map with tabId
+	const activeTab = TAB_MAP[tabId];
+	const navTabs = TABS.map(t => ({
+		...t,
+		active: tabId === t.id,
+		label: t.label + TAB_MAP[t.id].amount,
+	}));
+
+	// Render
+	const renderAddButton = () => {
+		if (activeTab.addHandler) {
+			return <Button icon="add" label="Aanmaken" onClick={activeTab.addHandler} type="secondary" />;
+		}
+	};
+
+	const renderFilter = () => {
+		const filter = activeTab.filter;
+
+		if (filter) {
+			if (!activeFilter) {
+				setActiveFilter(filter.options[0].id);
+			}
+
+			const currentFilter = filter.options.find(f => f.id === activeFilter);
+
+			return (
+				<Form type="inline">
+					<FormGroup label={filter.label}>
+						<ControlledDropdown isOpen={false} placement="bottom-end">
+							<DropdownButton>
+								<div className="c-filter-dropdown c-filter-dropdown--no-bg">
+									<div className="c-filter-dropdown__label">
+										{currentFilter ? currentFilter.label : filter.options[0].label}
+									</div>
+									<div className="c-filter-dropdown__options">
+										<Icon name="caret-down" />
+									</div>
+								</div>
+							</DropdownButton>
+							<DropdownContent>
+								<MenuContent
+									menuItems={filter.options}
+									onClick={filter => {
+										setActiveFilter(filter);
+									}}
+								/>
+							</DropdownContent>
+						</ControlledDropdown>
+					</FormGroup>
+				</Form>
+			);
+		}
+	};
+
 	return (
 		<Fragment>
-			<Tabs
-				tabs={[
-					{
-						label: 'Collections',
-						icon: 'collection',
-						id: 'collecties',
-						active: tabId === 'collecties',
-					},
-					{
-						label: 'Bookmarks',
-						icon: 'bookmark',
-						id: 'bladwijzers',
-						active: tabId === 'bladwijzers',
-					},
-				]}
-				onClick={goToTab}
-			/>
-			<br />
-			<span>TODO mijn werkruimte</span>
-			<br />
-			{(tabId === 'collecties' || !tabId) && (
-				<Collections location={location} history={history} match={match} />
-			)}
-			{tabId === 'bladwijzers' && <Bookmarks location={location} history={history} match={match} />}
+			<Container background="alt" mode="vertical" size="small">
+				<Container mode="horizontal">
+					<Toolbar autoHeight>
+						<ToolbarLeft>
+							<ToolbarItem>
+								<h2 className="c-h2 u-m-0">Mijn Archief</h2>
```

Comment:
design change: **Mijn archief** should be: **Mijn Werkruimte**

Reply:
fixed